moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h

```diff
@@ -143,13 +143,14 @@ class MoveItCpp
   const std::map<std::string, planning_pipeline::PlanningPipelinePtr>& getPlanningPipelines() const;
 
   /** \brief Get the stored instance of the planning scene monitor */
-  const planning_scene_monitor::PlanningSceneMonitorPtr& getPlanningSceneMonitor() const;
+  planning_scene_monitor::PlanningSceneMonitorConstPtr getPlanningSceneMonitor() const;
   planning_scene_monitor::PlanningSceneMonitorPtr getPlanningSceneMonitorNonConst();
 
-  const std::shared_ptr<tf2_ros::Buffer>& getTFBuffer() const;
+  std::shared_ptr<const tf2_ros::Buffer> getTFBuffer() const;
+  std::shared_ptr<tf2_ros::Buffer> getTFBuffer();
```

Comment:
I was never a fan of the `*NonConst` naming scheme and think this approach is much cleaner. :+1:

Reply:
Me neither. The *NonConst scheme was introduced/continued by @henningkayser. I didn't attempt to change that.